### DISCUSSION TITLE
Allow text fields to be searchable in an upload collection

### DISCUSSION THREAD

**Discussion by christian-reichart** [payloadcms/payload#974] [Feature Requests & Ideas]
It would be great if a collection could be configured so text fields are included in the full text search, rather than having to use the filter function.

Use case:
I'm creating a collection for an icon library and added a field called "tags", so that icons can be searched efficiently. E.g. a pencil icon could have the tag "edit".

It would be great for our editors to just use the text search.

Solution:
Extend the admin object of the collection:

```
admin {
  searchableTextFields: ['tags']
}
```

**Comment by jmikrut** [comment]
This is a beautiful idea.

We would welcome this as a PR. The code to adjust is [here](https://github.com/payloadcms/payload/blob/master/src/admin/components/elements/SearchFilter/index.tsx). We'd probably need to feed the collection config itself into this component as a prop, so that we could access the `searchableTextFields` and then create a `where` query containing a few `or` clauses (one for each field to be searched on), but this is doable!

If no one gets to this, we will eventually for sure as I really dig the idea.

**Reply by christian-reichart** [reply]
Awesome, I'll hopefully find the time soon to create a PR.
